ietfdata/mailarchive.py

```diff
@@ -51,6 +51,27 @@ def _parse_archive_url(archive_url:str) -> Tuple[str, str]:
 
 # =================================================================================================
 
+class MessageThread:
+    _msg_ids : List[str]
+    messages: List[Tuple[int, Message]]
+    
+    def __init__(self, index: int, first_message: Message):
+        self._msg_ids =  [first_message['Message-ID']]
+        self.messages = [(index, first_message)]
+
+        
+    def should_contain(self, msg: Message):
+        return msg["In-Reply-To"] in self._msg_ids
+
```

Comment:
What about the `References` header?

Reply:
Added supported for this.

Reply:
Not sure I understand this code - it looks like it only checks the first entry in `References` header, but surely needs to check all?

Reply:
Ah -- missing `if` statement: added.